src/tasks/task_list.hpp

```diff
@@ -142,6 +211,14 @@ class TaskList {
     return valid;
   }
 
+  TaskID AddTask(Task tsk) {
+    TaskID id(tasks_added_ + 1);
+    tsk.SetID(id);
+    task_list_.push_back(tsk);
+    tasks_added_++;
+    return id;
+  }
```

Comment:
Hmm I see. Now the task id needs to be kind of exposed and tracked. In principle I think this is fine.